consensus/hotstuff/safetyrules/safety_rules_test.go

```diff
@@ -447,29 +483,75 @@ func (s *SafetyRulesTestSuite) TestProduceTimeout_ShouldTimeout() {
 // Hence, the cases tested here would be symptoms of an internal bugs, and therefore should not result in an NoVoteError.
 func (s *SafetyRulesTestSuite) TestProduceTimeout_NotSafeToTimeout() {
 
-	s.Run("highest-qc-below-locked-round", func() {
-		view := s.proposal.Block.View
-		newestQC := helper.MakeQC(helper.WithQCView(s.safetyData.LockedOneChainView - 1))
+	s.Run("newest-qc-nil", func() {
+		// newestQC cannot be nil
+		timeout, err := s.safety.ProduceTimeout(s.safetyData.LockedOneChainView, nil, nil)
+		require.Error(s.T(), err)
+		require.False(s.T(), model.IsNoVoteError(err))
+		require.Nil(s.T(), timeout)
+	})
+	s.Run("happy-path-includes-last-view-tc", func() {
```

Comment:
```suggestion
	// if a QC for the previous view is provided, a last view TC is unnecessary and must not be provided
	s.Run("includes-last-view-qc-and-tc", func() {
```
Also not sure why this is named happy path, since we're testing an error condition